lighthouse-cli/test/smokehouse/lighthouse-runners/devtools.js

```diff
@@ -77,37 +61,19 @@ async function buildDevtools(logs) {
  * @return {Promise<{lhr: LH.Result, artifacts: LH.Artifacts, log: string}>}
  */
 async function runLighthouse(url, configJson, testRunnerOptions = {}) {
-  /** @type {string[]} */
-  const logs = [];
+  if (!buildDevtoolsPromise) buildDevtoolsPromise = buildDevtools();
+  await buildDevtoolsPromise;
 
-  if (!buildDevtoolsPromise) buildDevtoolsPromise = buildDevtools(logs);
-  if (!await buildDevtoolsPromise) {
```

Comment:
this bit was important before to prevent continuing with the smokes if previous smoke test dfns failed to build devtools ... is that not the case anymore?

It may be a good reason to have an actual "setup" callback for a runner now. If you check the PR history you'll see it was removed because it seemed like we could get equivalent behavior using a cached promise. but it seems to make error reporting (of the setup code) harder to reason about

Reply:
If build fails an error is thrown and the entire script quits.

Reply:
What do you mean "the script quits?" Smoke runner will retry n times and then continue on to the next test dfn, all of which will fail if the setup for the runner is failing

<img width="880" alt="image" src="https://user-images.githubusercontent.com/4071474/178595607-8987a827-16b0-4107-9fc8-d795afd5001f.png">

Reply:
btw am fine with doing a "runner setup" fn in a new PR, I'll take that on

Reply:
Clarification:
`buildDevtools` is only run once, but the output promise is awaited for every test * # of retries so the output error is pretty noisy.

Setup callback sounds good to me.